File: lab3/Tipper/app/src/main/java/com/example/tipper/MainMenuActivity.java
package com.example.tipper;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainMenuActivity extends AppCompatActivity {

    Button btnOpenCalculator;
    Button btnOpenKcalCalculator;
    Button btnOpenQuizActivity;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_menu);

        btnOpenCalculator = (Button)findViewById(R.id.btnOpenCalculator);
        btnOpenKcalCalculator = (Button)findViewById(R.id.btnOpenKcalCalculator);
        btnOpenQuizActivity = (Button)findViewById(R.id.btnOpenQuizActivity);

        //BMI Calculator
        btnOpenCalculator.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openCalculatorActivityOnButtonClick();
            }
        });

        //Kcal Calculator
        btnOpenKcalCalculator.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                openKcalCalculatorActivityOnButtonClick();
            }
        });

        //Quiz
        btnOpenQuizActivity.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                btnOpenQuizActivityOnButtonClick();
            }
        });
    }

    //BMI method
    private void openCalculatorActivityOnButtonClick() {
        Intent calculatorActivity = new Intent(this, BMICalculatorActivity.class);
        startActivity(calculatorActivity);
    }

    //kcal method
    private void openKcalCalculatorActivityOnButtonClick() {
        Intent kcalCalculatorActivity = new Intent(this, KCalsCalculatorActivity.class);
        startActivity(kcalCalculatorActivity);
    }

    private void btnOpenQuizActivityOnButtonClick() {
        Intent quizActivity = new Intent(this, QuizActivity.class);
        startActivity(quizActivity);
    }
}

File: lab4/BMI_App/app/src/main/java/com/example/bmiapp/MainMenuActivity.kt
package com.example.bmiapp

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainMenuActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_menu)

        val btnOpenCalculator = findViewById<Button>(R.id.btnOpenCalculator)
        val btnOpenKcalCalculator = findViewById<Button>(R.id.btnOpenKcalCalculator)
        val btnOpenQuizActivity = findViewById<Button>(R.id.btnOpenQuizActivity)
        val btnQuitApp = findViewById<Button>(R.id.btnQuitApp)

        btnOpenCalculator.setOnClickListener{
            val calculatorActivity  = Intent(this, BMICalculatorActivity::class.java)
            startActivity(calculatorActivity)
        }

        btnOpenKcalCalculator.setOnClickListener{
            val kcalCalculatorActivity = Intent(this, KCalsCalculatorActivity::class.java)
            startActivity(kcalCalculatorActivity)
        }

        btnOpenQuizActivity.setOnClickListener{
            val quizActivity = Intent(this, QuizActivity::class.java)
            startActivity(quizActivity)
        }

        btnQuitApp.setOnClickListener {
            finish()
        }
    }
}